File: platforms/python/examples/core/renderer/update_texture_with.py
from fragmentcolor import Renderer, TextureFormat, TextureWriteOptions

renderer = Renderer()
texture = renderer.create_storage_texture([64, 32], TextureFormat.Rgba, None)
id = texture.id()
frame = bytes(64 * 32 * 4)
opt = TextureWriteOptions.whole().with_bytes_per_row(256).with_rows_per_image(32)

renderer.update_texture_with(id, frame, opt)

File: platforms/swift/examples/core/renderer/update_texture_with.swift
import FragmentColor
let renderer = Renderer()
let texture = try await renderer.createStorageTexture([64, 32], TextureFormat.Rgba, nil)
let id = *texture.id()
let frame = [0u8; 64 * 32 * 4]
let opt = TextureWriteOptions.whole().withBytesPerRow(256).withRowsPerImage(32)

renderer.updateTextureWith(id, frame, opt)